modules/standard/IO.chpl

```diff
@@ -313,26 +313,24 @@ input, standard output, and standard error.
 :var:`stdout` and :var:`stderr` support writing.
 
 All three channels are safe to use concurrently.
-Their types' ``kind`` argument is ``dynamic``
+Their types' ``kind`` argument is ``dynamic``.
 
 .. _about-io-error-handling:
 
 Error Handling
 --------------
 
 Most I/O routines throw a :class:`SysError.SystemError`, and can be handled
-appropriately with ``try`` and ``catch``. For legacy reasons most I/O routines
+appropriately with ``try`` and ``catch``. For legacy reasons, most I/O routines
 can also can accept an optional `error=` argument.
```

Comment:
I think we've removed the `error=` versions a this point so that sentence should also be removed. It would be nice to remove other comments referring to the `error= version`.

Reply:
These procedures still take a `ref error`, even if they are `#pragma "no doc"` (all but the last one):

```
pragma "no doc"
proc channel._match_regexp_if_needed(cur:size_t, len:size_t, ref error:syserr,
    ref style:iostyle, r:unmanaged _channel_regexp_info)
{
}

pragma "no doc"
proc channel._format_reader(
    fmtStr:?fmtType, ref cur:size_t, len:size_t, ref error:syserr,
    ref conv:qio_conv_t, ref gotConv:bool, ref style:iostyle,
    ref r:unmanaged _channel_regexp_info?,
    isReadf:bool)
{
}

pragma "no doc"
proc channel._conv_helper(
    ref error:syserr,
    ref conv:qio_conv_t, ref gotConv:bool,
    ref j:int,
    argType: c_ptr(c_int),
    argTypeLen: int)
{
}

pragma "no doc"
proc channel._conv_sethandler(
    ref error:syserr,
    argtypei:c_int,
    ref style:iostyle,
    i:int, argi,
    isReadf:bool):bool throws
{
}

pragma "no doc"
proc channel._extractMatch(m:reMatch, ref arg:reMatch, ref error:syserr) {
  // If the argument is a match record, just return it.
  arg = m;
}

pragma "no doc"
proc channel._extractMatch(m:reMatch, ref arg:bytes, ref error:syserr) {
}

pragma "no doc"
proc channel._extractMatch(m:reMatch, ref arg:?t, ref error:syserr)
      where t != reMatch && t != bytes {
}

// documented in throws version
pragma "no doc"
proc channel.extractMatch(m:reMatch, ref arg, ref error:syserr) {
}

pragma "no doc"
proc channel._ch_handle_captures(matches:_ddata(qio_regexp_string_piece_t),
                                 nmatches:int,
                                 ref captures, ref error:syserr) {
}

pragma "no doc"
proc channel.search(re:regexp(?), ref error:syserr):reMatch
{
}

pragma "no doc"
proc channel.match(re:regexp(?), ref error:syserr):reMatch
{
}

proc channel.match(re:regexp(?), ref captures ...?k, ref error:syserr):reMatch
{
}
```

Reply:
Right, since these are almost all no-doc we don't need to describe them in the module docs.

The last one should get a deprecation warning if it doesn't already have one.

Reply:
What I really mean is - some of these are internal (the ones beginning with `_`) and those we can leave as -is. The others should get a deprecation warning.

Reply:
Some of them simply appear to be overloads with 0 captures and with >0 captures, with the 0-capture ones being `pragma "no doc"`. This is all to get around the limitation that varargs can't work with 0 arguments.

Some of them are still used by some of the other modules. For the scope of this PR, I don't want to have to deprecate and refactor the other modules' usage.

Please see commit https://github.com/chapel-lang/chapel/pull/17448/commits/8444714fad9bdd805d08be5e114068cedf6ddb53